### PR TITLE
ref(ui): Extract DiamondStatus component

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This is used in alerts, and will also be used in crons

### CODE REVIEW

**Review by k-fish — APPROVED** [review]

**Review by scttcper — APPROVED** [review]

File: static/app/components/diamondStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+function DiamondStatus({color, icon: Icon, ...props}: DiamondStatusProps) {
+  return (
+    <StatusWrapper role="presentation" color={color} {...props}>
+      <DiamondBackground color={color} />
+      <Icon color="white" />
+    </StatusWrapper>
+  );
+}
+
+export {DiamondStatus};
```
Comment (priscilawebdev) [inline static/app/components/diamondStatus.tsx:30]: would it be better to have the export near the function, for instance, `export function DiamondStatus`?

File: static/app/components/diamondStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  height: 36px;
+`;
+
+const StatusWrapper = styled('div')<{color: ColorOrAlias}>`
+  width: 36px;
+  height: 36px;
+  position: relative;
+
+  svg:last-child {
+    width: '16px';
```
Comment (priscilawebdev) [inline static/app/components/diamondStatus.tsx:43]:
```suggestion
    width: 16px;
```

**Comment by priscilawebdev** [comment]
What does this diamond look like? shouldn't we add a visual snapshot test?

**Comment by evanpurkhiser** [comment]
> What does this diamond look like? shouldn't we add a visual snapshot test?

I think we're okay, this is a pretty simple composition component, we already have higher level acceptance tests on the usages of it.

File: static/app/components/diamondStatus.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/components/diamondStatus.tsx:30]: we should decide on some standard here, that's a good question!

**Pull request merged**